src/renderers/dom/shared/inputValueTracking.js

```diff
@@ -128,32 +128,41 @@ var inputValueTracking = {
     inst._wrapperState.valueTracker = trackValueOnNode(node, inst);
   },
 
+  // TODO: in practice "subject" can currently be Stack instance,
+  // Fiber, or DOM node. This is hard to understand. We should either
+  // make it accept only DOM nodes, or only Fiber/Stack instances.
   updateValueIfChanged(subject: SubjectWithWrapperState | Fiber) {
     if (!subject) {
       return false;
     }
-    var tracker = getTracker(subject);
 
+    var isNode = (subject: any).nodeType === ELEMENT_NODE;
+    var tracker = getTracker(subject);
     if (!tracker) {
-      if (typeof (subject: any).tag === 'number') {
+      if (isNode) {
+        // DOM node
+        inputValueTracking.trackNode((subject: any));
+      } else if (typeof (subject: any).tag === 'number') {
+        // Fiber
         inputValueTracking.trackNode((subject: any).stateNode);
       } else {
+        // Stack
         inputValueTracking.track((subject: any));
       }
       return true;
     }
 
-    var lastValue = tracker.getValue();
-
-    var node = subject;
-
-    // TODO: remove check when the Stack renderer is retired
-    if ((subject: any).nodeType !== ELEMENT_NODE) {
+    var node;
+    if (isNode) {
+      // DOM node
+      node = subject;
+    } else {
+      // Fiber and Stack
       node = ReactDOMComponentTree.getNodeFromInstance(subject);
```

Comment:
It should, yeah. See [ReactDOMComponentTree.js#L190-L194](https://github.com/facebook/react/blob/b1768b5a48d1f82e4ef4150e0036c5f846d3758a/src/renderers/dom/shared/ReactDOMComponentTree.js#L190-L194)